src/notifications.ts

```diff
@@ -210,6 +210,10 @@ export class Notifications {
       if (handlerIndex > -1) {
         handlers.splice(handlerIndex, 1);
       }
+      if (!handlers.length) {
+        this.subscriptions[t] = [];
+        delete this.bulkSubscriptions[t];
```

Comment:
I think I found the root cause. I will try to leave a more detailed explanation, but for clarity, there are two issues with this approach: 

1. By only setting `this.subscriptions[t]` to an empty array, won't actually remove the sub from any future bulk subscribes (more on that in my in detailed explanation). But you will actually want to just delete the whole property
2. We don't want to delete `this.bulkSubscriptions[t]` because it is possible another consumer subscribed to the same topic using the bulk subscribe instead of an individual topic. Unsubscribing from individual topics should not remove bulk subscriptions. 

#### More on bullet 2
The bug is actually in the `resubscribe()` function. SC will eventually treat _all_ individual topic subscriptions as bulk subscriptions. That is why when you unsubscribe from an individual sub, sc never actually removes it because it _also thinks_ that it is a bulk sub so it don't remove it. 

If we can correct the `resubscribe()` function to know the difference between bulk subs and individual subs, we can use part of this to make sure we don't keep subs hanging around. 

I appreciate your work on this, @shanebruggeman. I will open a PR with my fix and I'll have you look it over to see if it fixes your issue.

Reply:
Appreciate the review @djhouseknecht! I'm more than willing to review a better change - this is just what I cobbled together based off following the subscribe / unsubscribe logic. And feel free to close this when it makes sense!